src/modules/iotjs_module_buffer.c

```diff
@@ -411,12 +409,12 @@ JHANDLER_FUNCTION(Slice) {
   int length = end - start;
   IOTJS_ASSERT(length >= 0);
 
-  iotjs_jval_t jnew_buffer = iotjs_bufferwrap_create_buffer(length);
+  iotjs_jval_t jnew_buffer = iotjs_bufferwrap_create_buffer((size_t)length);
   iotjs_bufferwrap_t* new_buffer_wrap =
       iotjs_bufferwrap_from_jbuffer(&jnew_buffer);
   iotjs_bufferwrap_copy_internal(new_buffer_wrap,
-                                 iotjs_bufferwrap_buffer(buffer_wrap), start,
-                                 end, 0);
+                                 iotjs_bufferwrap_buffer(buffer_wrap),
+                                 (size_t)start, (size_t)end, 0);
```

Comment:
Considering `bound_range` semantics, yes I think you're right. Then we can trick using `SIZE_MAX`. the first argument of `bound_range` is used as an index, so it can't be `SIZE_MAX`.(Because the length is less than `SIZE_MAX`). So inside the `bound_range`, we can implement like below.
```c
static size_t bound_range(size_t index, size_t low, size_t upper) {
  if (index == SIZE_MAX) {
    return low;
  }  
  if (index > upper) {
    return upper;
  }
  return index;
}
```
Then with few preprocessing, for example here` 
```c
int start = JHANDLER_GET_ARG(0, number);
size_t start_idx;

if (start < 0 && -start > len) {
  start_idx = SIZE_MAX;
} else {
  start_idx = start;
}

start_idx = bound_range(start_idx, 0, iotjs_bufferwrap_length(buffer_wrap));
```
What do you think ??